bn_mp_set_double.c

```diff
@@ -41,19 +54,119 @@ int mp_set_double(mp_int *a, double b)
       return res;
    }
 
-   if (((cast.bits >> 63) != 0ULL) && !IS_ZERO(a)) {
-      a->sign = MP_NEG;
+   if (((cast.bits >> PAYLOAD) != 0ULL) && (mp_iszero(a) == MP_NO)) {
+      SIGN(a) = MP_NEG;
```

Comment:
I think @fperrad removed all the usage of the SIGN macro etc. Basically you reverted @fperrad's changes.

Reply:
I thought that, too, was unsure, grepped for it, found a large bunch of&hellip;oh, they are all in mpi.c?